opslevel/datasource_opslevel_filter.go

```diff
@@ -54,10 +54,13 @@ func filterFilters(data []opslevel.Filter, field string, value string) (*opsleve
 
 func datasourceFilterRead(d *schema.ResourceData, client *opslevel.Client) error {
 	resp, err := client.ListFilters(nil)
-	results := resp.Nodes
 	if err != nil {
 		return err
 	}
+	if resp == nil {
+		return fmt.Errorf("unexpected: listing filters returned nil")
+	}
+	results := resp.Nodes
```

Comment:
what about if `resp.Nodes` is empty?

Reply:
@rocktavious `resp.Nodes` here is only being used for `range` which is why `nilaway` only warned me about line 57 in the old code. Therefore this nil check is sufficient. [(link to usage of `resp.Nodes`)](https://github.com/OpsLevel/terraform-provider-opslevel/blob/920c247641f67d1fdcf431d9f2cc4bbf3b98239a/opslevel/datasource_opslevel_filter.go#L31)

In GoLang arrays and maps are interfaces and doing `range` or even `len` over them will not cause a nil pointer panic. [Demonstration](https://go.dev/play/p/_GWLMuyxcLh)

If someone does make an update to do something than other `range` over the `resp.Nodes`, it will trigger `nilaway` and force them to add nil checks.